types/multi_address.go

```diff
@@ -60,8 +60,10 @@ func (m MultiAddress) Encode(encoder scale.Encoder) error {
 	var err error
 	switch {
 	case m.IsID:
-		if err = encoder.PushByte(0); err != nil {
-			return err
+		if !defaultOptions.NoPalletIndices { // Skip in case target chain doesn't include indices pallet
```

Comment:
Does this only apply for the first enum which is `AccountID`? What about the rest?

Reply:
If I understand correctly, you are asking for a reference to a documentation, specification or a canonical implementation. I haven't found this reference yet. The patch was created in this form because 1) this is how it is done for the type `Address` in `Encode` [types/address.go](https://github.com/centrifuge/go-substrate-rpc-client/blob/master/types/address.go#L116) resp. `Decode` [types/address.go](https://github.com/centrifuge/go-substrate-rpc-client/blob/master/types/address.go#L66) 2) omitting the byte in this particular case makes the code work, i.e. submitting a signed extrinsic to HydraDX is successful this way.

Unfortunately there is no reference to motivation for this piece of code in the `Address` type or in the introducing commit and I've so far had no luck in finding the specification. In the ideal world this would probably be part of the metadata, but I don't see it there.